src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,236 @@
+    setSelectedCentury([15, 16, 17, 18, 19]);
+  };
+
+  const handleSort = (column: string) => {
+    setPropName(column);
+    setClickCount(clickCount + 1);
+    if (propName === column) {
+      if (clickCount === 1) {
+        setSortOrder(SortType.asc);
+      }
+
+      if (clickCount === 2) {
+        setSortOrder(SortType.desc);
+      }
+    }
+
+    if (clickCount === 3) {
+      setPropName(column);
+      setClickCount(1);
+      setSortOrder(SortType.og);
+    }
+  };
+
+  const resetEveryThing = () => {
+    setSelectedCentury([15, 16, 17, 18, 19]);
+    setQuery('');
+    sexFilterHandler(FilterType.All);
+  };
+
+  const setCurrentQuery = (currentQuery: string) => {
+    setQuery(currentQuery);
+  };
+
+  const handleCenturySelection = (century : number) => {
+    if (selectedCentury.includes(century) && selectedCentury.length === 5) {
+      const filtered = selectedCentury.filter((number) => {
+        return number === century;
+      });
+
+      setSelectedCentury(filtered);
+    } else if (!selectedCentury.includes(century)
+    && selectedCentury.length !== 5) {
+      setSelectedCentury([...selectedCentury, century]);
+    } else if (selectedCentury.includes(century)) {
+      const filtered = selectedCentury.filter((number) => {
+        return number !== century;
+      });
+
+      setSelectedCentury(filtered);
+    }
+  };
+
+  const updatedPeople = people.map((child) => {
+    if (!child.motherName && !child.fatherName) {
+      return {
+        ...child,
+        motherName: '-',
+        fatherName: '-',
+      };
+    }
+
+    if (!child.fatherName) {
+      return {
+        ...child,
+        fatherName: '-',
+      };
+    }
+
+    if (!child.motherName) {
+      return {
+        ...child,
+        motherName: '-',
+      };
+    }
+
+    const father = people.find((parent) => parent.name === child.fatherName);
+    const mother = people.find((parent) => parent.name === child.motherName);
+
+    return {
+      ...child,
+      father,
+      mother,
+    };
+  });
+
+  const filteredPeople: NewPerson[] = useMemo(
+    () => updatedPeople.filter((person) => {
+      const functionality
+       = person.name.toLowerCase().includes(query.toLowerCase().trim());
+      const centurie = Math.floor(person.born / 100);
+
+      switch (sexFilter) {
+        case FilterType.All:
+        default:
+          return functionality && selectedCentury.includes(centurie);
+        case FilterType.Male:
+          return (
+            person.sex === 'm'
+            && functionality
+            && selectedCentury.includes(centurie)
+          );
+        case FilterType.Female:
+          return (
+            person.sex === 'f'
+            && functionality
+            && selectedCentury.includes(centurie)
+          );
+      }
+    }),
+    [sexFilter, query, loading, selectedCentury],
+  );
+
+  useEffect(() => {
+    setIsfiltered(true);
+  }, [query, sexFilter, selectedCentury]);
```

Comment:
Use a boolean to track whether the component is filtered or not may cause unexpected behavior. Consider using a function to check if the filters are in their initial state.